src/compiler/parser.ts

```diff
@@ -440,7 +440,26 @@ namespace ts {
 
     /* @internal */
     export function parseIsolatedJSDocComment(content: string, start?: number, length?: number) {
-        return Parser.JSDocParser.parseIsolatedJSDocComment(content, start, length);
+        const result = Parser.JSDocParser.parseIsolatedJSDocComment(content, start, length);
+        if (result.jsDocComment) {
+            // because the jsDocComment was parsed out of the source file, it might
+            // not be covered by the fixupParentReferences.
+            let parentNode: Node = result.jsDocComment;
+            forEachChild(result.jsDocComment, visitNode);
+
+            function visitNode(n: Node): void {
+                if (n.parent !== parentNode) {

```

Comment:
I'd expect the check here to be `n.parent === undefined`. Is there ever a case where the node has a parent but it's not the current parent? That seems like it should never happen.